src/runner.py

```diff
@@ -17,31 +32,40 @@ def main() -> int:
         help="Configuration location",
     )
     parser.add_argument("interval", help="Time in seconds between runs", type=int)
+    parser.add_argument(
+        "--hours-range",
+        default="0-23",
+        help="Hours of day interval expresses as 0-23 or 2",
+        type=parseNumList,
+    )
     args = parser.parse_args()
 
     print(f"Running bandersnatch every {args.interval}s", file=sys.stderr)
     try:
         while True:
-            start_time = time()
-
-            try:
-                cmd = [
-                    sys.executable,
-                    "-m",
-                    "bandersnatch.main",
-                    "--config",
-                    args.config,
-                    "mirror",
-                ]
-                run(cmd, check=True)
-            except CalledProcessError as cpe:
-                return cpe.returncode
-
-            run_time = time() - start_time
-            if run_time < args.interval:
-                sleep_time = args.interval - run_time
-                print(f"Sleeping for {sleep_time}s", file=sys.stderr)
-                sleep(sleep_time)
+            if datetime.now().hour in args.hours_range:
+                start_time = time()
+
+                try:
+                    cmd = [
+                        sys.executable,
+                        "-m",
+                        "bandersnatch.main",
+                        "--config",
+                        args.config,
+                        "mirror",
+                    ]
+                    run(cmd, check=True)
+                except CalledProcessError as cpe:
+                    return cpe.returncode
+
+                run_time = time() - start_time
+                if run_time < args.interval:
+                    sleep_time = args.interval - run_time
+                    print(f"Sleeping for {sleep_time}s", file=sys.stderr)
+                    sleep(sleep_time)
+            else:
+                sleep(60)
```

Comment:
```suggestion
                sleep(args.interval)
```
Why default this to 60 and not use the supplied interval?

Reply:
I used a fixed 60 seconds there to avoid some impossible configurations. Like run with 7200 seconds and just a single hour in the provided interval. With 60 at least I'm sure it will check the current time every minute and run bandersnatch in every condition.

I'm sure there would better ways to do that, that was my idea in doing that.

Reply:
I wanted to add tests for this, but I'm not an expert in mocking datetime and sleeps together. Would be nice to test that, too...

Reply:
I can add these. Thanks for making some tests. Much appreciated.